File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, nums: List[int]) -> int:
        i=0
        j=1
        max_num = 0
        def operation(i,j,max_num):
            if i < len(nums) and j < len(nums):
                if nums[i]>nums[j]:
                    i = j
                    j = j+1
                    return operation(i,j,max_num)
                else:
                    diff = nums[j] - nums[i]
                    if diff > max_num:
                        max_num = diff
                    j = j+1
                    return operation(i,j,max_num)
            else:
                return max_num

        return operation(i,j,max_num)

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let i = 0;
    let j = 1;
    let max = 0;

    const recursion = (i, j) => {
        while (j < prices.length) {
            const diff = prices[j] - prices[i];
            if (diff > max) {
                max = diff
            }

            if (prices[i] > prices[j]) {
                i = j
            }
            j++

        }
    }
    recursion(i, j)
    return max
};